components/workflow-mgt/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/listener/AbstractWorkflowImplServiceListener.java

```diff
@@ -0,0 +1,216 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.workflow.impl.listener;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.core.model.IdentityEventListener;
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.workflow.impl.WorkflowImplException;
+import org.wso2.carbon.identity.workflow.impl.bean.BPSProfile;
+import org.wso2.carbon.identity.workflow.mgt.bean.Workflow;
+import org.wso2.carbon.identity.workflow.mgt.bean.WorkflowRequest;
+
+public abstract class AbstractWorkflowImplServiceListener implements WorkflowImplServiceListener {
+
+    /**
+     * Trigger before adding new BPS profile
+     *
+     * @param bpsProfileDTO
+     * @param tenantId
+     * @throws WorkflowImplException
+     */
+    @Override
+    public void doPreAddBPSProfile(BPSProfile bpsProfileDTO, int tenantId) throws WorkflowImplException {
+
+    }
+
+    /**
+     * Trigger after adding new BPS profile
+     *
+     * @param bpsProfileDTO
+     * @param tenantId
+     * @throws WorkflowImplException
+     */
+    @Override
+    public void doPostAddBPSProfile(BPSProfile bpsProfileDTO, int tenantId) throws WorkflowImplException {
+
+    }
+
+    /**
+     * Trigger before listing BPS profiles
+     *
+     * @param tenantId
+     * @throws WorkflowImplException
+     */
+    @Override
+    public void doPreListBPSProfiles(int tenantId) throws WorkflowImplException {
+
+    }
+
+    /**
+     * Trigger after listing BPS profiles
+     *
+     * @param tenantId
+     * @throws WorkflowImplException
+     */
+    @Override
+    public void doPostListBPSProfiles(int tenantId) throws WorkflowImplException {
+
+    }
+
+    /**
+     * Trigger before removing a BPS profile
+     *
+     * @param profileName
+     * @throws WorkflowImplException
+     */
+    @Override
+    public void doPreRemoveBPSProfile(String profileName) throws WorkflowImplException {
+
+    }
+
+    /**
+     * Trigger after removing a BPS profile
+     *
+     * @param profileName
+     * @throws WorkflowImplException
+     */
+    @Override
+    public void doPostRemoveBPSProfile(String profileName) throws WorkflowImplException {
+
+    }
+
+    /**
+     * Trigger before retrieving a BPS profile
+     *
+     * @param profileName
+     * @param tenantId
+     * @throws WorkflowImplException
+     */
+    @Override
+    public void doPreGetBPSProfile(String profileName, int tenantId) throws WorkflowImplException {
+
+    }
+
+    /**
+     * Trigger after retrieving a BPS profile
+     *
+     * @param profileName
+     * @param tenantId
+     * @throws WorkflowImplException
+     */
+    @Override
+    public void doPostGetBPSProfile(String profileName, int tenantId) throws WorkflowImplException {
```

Comment:
In this one and all the workflow listeners, for any get methods' post listeners, can you pass the result of the do method also as the first final argument.